source/proximity_list/proximitylist_api.cpp

```diff
@@ -126,16 +127,18 @@ void find(navitia::PbCreator& pb_creator,
                 pb_creator.fill_pb_error(pbnavitia::Error::bad_filter, "ptref : " + pt_error.more);
                 return;
             }
+            // We have to find all objects within distance, then apply the filter
+            search_count = -1;
         }
         switch (type) {
             case nt::Type_e::StopArea:
-                list = pb_creator.data->pt_data->stop_area_proximity_list.find_within(coord, distance, count);
+                list = pb_creator.data->pt_data->stop_area_proximity_list.find_within(coord, distance, search_count);
```

Comment:
Can't you modify stop_area_proximity_list before? (copy then filter it?).
Or maybe pass a visitor to filter into radius_search?

Reply:
Since the performance is quite acceptable so far, i resolve this conversation